epochStart/bootstrap/epochStartDataProvider.go

```diff
@@ -0,0 +1,103 @@
+package bootstrap
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+)
+
+var log = logger.GetOrCreate("registration")
+var _ process.Interceptor = (*simpleMetaBlockInterceptor)(nil)
+
+const requestSuffix = "_REQUEST"
+
+type metaBlockInterceptorHandler interface {
+	process.Interceptor
+	GetAllReceivedMetaBlocks() []block.MetaBlock
+}
+
+type shardHeaderInterceptorHandler interface {
+	process.Interceptor
+	GetAllReceivedShardHeaders() []block.ShardData
+}
+
+type EpochStartDataProvider struct {
+	marshalizer            marshal.Marshalizer
+	messenger              p2p.Messenger
+	metaBlockInterceptor   metaBlockInterceptorHandler
+	shardHeaderInterceptor shardHeaderInterceptorHandler
+}
+
+func NewEpochStartDataProvider(messenger p2p.Messenger, marshalizer marshal.Marshalizer) *EpochStartDataProvider {
+	metaBlockInterceptor := NewSimpleMetaBlockInterceptor(marshalizer)
+	shardHdrInterceptor := NewSimpleShardHeaderInterceptor(marshalizer)
+	return &EpochStartDataProvider{
+		marshalizer:            marshalizer,
+		messenger:              messenger,
+		metaBlockInterceptor:   metaBlockInterceptor,
+		shardHeaderInterceptor: shardHdrInterceptor,
+	}
+}
+
+func (ser *EpochStartDataProvider) RequestEpochStartMetaBlock(epoch uint32) (*block.MetaBlock, error) {
+	err := ser.messenger.CreateTopic(factory.MetachainBlocksTopic+requestSuffix, false)
+	if err != nil {
+		return nil, err
+	}
+
+	err = ser.messenger.CreateTopic(factory.MetachainBlocksTopic, false)
```

Comment:
this should be called with true (create a dedicated pipe) Why not reusing the code from interceptors factory that creates the interceptor for this topic?

Reply:
done